packages/signature-controller/src/types.ts

```diff
@@ -71,11 +77,29 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
+  /** Signature method V1, V3 or V4 */
+  signatureMethod?: string;
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+export type DecodedRequestInfo =
```

Comment:
Should we avoid coupling to the API and call this `DecodingData` as a parallel to `SimulationData` in the transaction controller?

Reply:
Fixed.